google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/gcs/GcsBucketPanel.java

```diff
@@ -124,14 +115,16 @@ void refresh() {
 
     if (!Services.getLoginService().isLoggedIn()) {
       notificationLabel.setText(GctBundle.message("gcs.panel.bucket.listing.not.logged.in"));
-    } else if (StringUtils.isEmpty(projectSelector.getText())) {
+    } else if (projectSelector.getSelectedProject() == null) {
       notificationLabel.setText(GctBundle.message("gcs.panel.bucket.listing.no.project.selected"));
     } else {
-      String projectId = projectSelector.getText();
-      CredentialedUser user = projectSelector.getSelectedUser();
+      CloudProject cloudProject = projectSelector.getSelectedProject();
+      String projectId = cloudProject.projectId();
```

Comment:
is this only used once? if so could you get rid of this variable extraction?

Reply:
We also grab google user name from it

Reply:
Or you mean remove projectId?

Reply:
i meant getting rid of line 122 `String projectId = cloudProject.getProjectId()` since its only used once; you could just call `cloudProject.projectId` directly on line 127 no?

Reply:
right, sounds good.

Reply:
Also, I'm noticing the naming of the methods in CloudProject aren't standard pojo naming: `CloudProject#projectId` instead of `CloudProject#getProjectId`. Does this have something to do with `@AutoValue`

Reply:
Yes, it looks like that the default way AutoValue is used: https://g3doc.corp.google.com/third_party/java_src/auto/value/g3doc/index.md?cl=head

Reply:
sounds good